### DISCUSSION TITLE
Support better HTML compression using htmlnano

### DISCUSSION THREAD

**Discussion by aminya** [withastro/roadmap#703] [Proposal]
# Summary

Support htmlnano for better HTML compression. 

# Background & Motivation

The astro compiler's `compressHTML` isn't enough:
- It doesn't remove comments from the output
- The inline CSS is not minified (e.g. when inserted by Astro plugins)

Also, using `posthtml` and `htmlnano` through the vite plugins doesn't work. 

```js
// https://astro.build/config

import posthtml from "@vituum/vite-plugin-posthtml"
import { defineConfig } from "astro/config"
import htmlNano from "htmlnano"

const browserlist = ["chrome71"]

export default defineConfig({
  site: "https://test.com",
  base: "/",
  trailingSlash: "always",
  vite: {
    target: browserlist,
    plugins: [
      posthtml({
        include: false,
        extend: false,
        plugins: [htmlNano()],  // doesn't do anything
      }),
    ],
  },
})

```
# Goals

A **concise, bulleted-list** outlining the intended goals of this RFC. 

- Support `htmlnano` 

# Example

See above

**Comment by aminya** [comment]
I published two astro-plugins to fix this issue. Now, `htmlnano` and `posthtml` are both supported for Astro!

https://www.npmjs.com/package/astro-htmlnano
https://www.npmjs.com/package/astro-posthtml

**Comment by cdtut** [comment]
Should be included by default or as option without external plugins. Only CSS and JS is minified properly but HTML is not is not consistent.

Maybe configuration option like this to pass options to htmlnano.

```js
htmlModify: {
  removeComments: 'safe'
}
```

Or disable minifying.


```js
htmlModify: false
```

**Reply by aminya** [reply]
Did you check astro-htmlnano?

https://github.com/withastro/roadmap/discussions/703#discussioncomment-7287261
